Bump PHP version to 7 and add typehinting

slax0rr opened the pull request:
As #4 requested, these changes bump the version in the composer.json file, and add typehints to all the classes and interfaces.

The tests are unfortunately now broken, since your curl_* mocks are more often than not returning strings instead of bool or int values as the originals.

slax0rr committed:
bump version in composer.json

slax0rr committed:
add typehinting to the classes and interfaces

slax0rr committed:
fix minor code styling errors

f3ath commented:
Thanks for your contribution. I imagine, tests should be relatively easy to fix too. Though it may be somewhat tedious. 
Travis config should also be updated, I think.

slax0rr commented:
Tests can be fixed quite easily yes, but I am currently swamped. Will handle them later this afternoon. Will update this PR.

slax0rr committed:
remove 5.* php versions from travis config

slax0rr committed:
return values from mocks that fit the new typehints

slax0rr committed:
attempt to fix travis hhvm php7 tests

slax0rr committed:
ignore platform reqs when testing with hhvm

slax0rr committed:
revert changes to travis config and ignore platform reqs

slax0rr committed:
another attempt for hhvm php7 travis tests

slax0rr committed:
keep "regular" php versions in "old" travis config format

slax0rr committed:
added hhvm to allowed failures

The HHVM team announced that HHVM will never truly support PHP7, in the
light of this I have added the hhvm to the allowed failures list in
traves. Source: http://hhvm.com/blog/2017/09/18/the-future-of-hhvm.html

slax0rr committed:
fix allow_failures error in travis config

f3ath commented:
Great job, thank you!
